ArduPlane/control_modes.cpp

```diff
@@ -73,6 +73,13 @@ Mode *Plane::mode_from_mode_num(const enum Mode::Number num)
     case Mode::Number::TAKEOFF:
         ret = &mode_takeoff;
         break;
+#if SOARING_ENABLED == ENABLED
+    case Mode::Number::THERMAL:
+        ret = &mode_thermal;
+        break;
+#endif
+    default:
```

Comment:
There shouldn't be a `default:` clause here.... put the `#if inside the `case` for thermal.

Reply:
Thanks, fixed.